packages/components/inputs/date-time-input/src/time-input.tsx

```diff
@@ -1,6 +1,8 @@
-import PropTypes from 'prop-types';
+/// <reference types="@emotion/react/types/css-prop" />
+import type { KeyboardEventHandler, RefObject } from 'react';
 import { css } from '@emotion/react';
 import { customProperties as vars } from '@commercetools-uikit/design-system';
+import type { TDateTimeInputProps } from './date-time-input';
```

Comment:
Not sure about this one as we're creating a circular dependency here.
I think we should invert the types dependencies. I mean, `DateTimeInput` should depend on `TimeInput` but the other way.

I guess `isDisabled`, `onChange`, `value` and `placeholder` should be defined here and imported from `DateTimeInput`.

Reply:
I think import type help solve this problem.
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html

Reply:
Thanks for the link, @Rhotimee.

I guess `import type` is for helping when dealing with types when you use a compiler like `Babel` which works one-file-at-a-time and you have the **isolatedModules** option set to `true` (this is our case) in your `tsconfig` file.

I realized this is not going to be a problem at build time because types are actually removed (they don't exist at runtime). However, I think conceptually `TimeInput` is the building block for `DateTimeInput` so the dependency should only go from the latter to the former. I think it's ok for the `DateTimeInput` to be coupled to `TimeInput` but I don't think the same for the opposite.
For me `TimeInput` is a building block and should not depend on other components.
([reference](https://bradfrost.com/blog/post/atomic-web-design/))

Actually, the props in the `DateTimeInput` have a different purpose so, overall, I don't think we should share them just because they match and this component should have its own (I wouldn't say the same if `DateTimeInput` had that props only to pass them through to the `TimeInput`).

Anyway, this is not a blocker.

Reply:
We had a meeting to discuss this and I realized I was confused about this file.
@Rhotimee explained the rationale so now I understand why it's done like this.